### PR TITLE
Implement Information box and ens banner view

### LINKED ISSUE #13743: Add banner for ENS redistribution
# Problem
We need more people to be aware that they can collect $ENS if they registered a stateofus.eth name.

## Implementation
see [Figma](https://www.figma.com/file/vMndFfhSLNNA7lknaHX5Ue/ENS-registration?node-id=8%3A18365)

Link `Open dApp` should open https://ens-collect.status.im/ in Status browser

- Banner should be shown on all profiles when opening a profile for the first time on release 1.20
- If dismissed on any profile, it's dismissed on all profiles

ENS redistribution banner

## Notes
Copy to be confirmed. Please ping @hesterbruikman

### LINKED ISSUE #13749: UI Component - Info Message and Information box
This issue is to track the creation of the component: Information box

**Info Message**: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=2950%3A43503
**Information box**: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=2950%3A58462

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes: https://github.com/status-im/status-mobile/issues/13749
fixes: https://github.com/status-im/status-mobile/issues/13743

### Summary
PR implements Info Message and Information Box component, and also adds ENS Banner for release 1.20

status: ready

### CODE REVIEW

File: src/quo2/foundations/typography.cljs
Comment (Parveshdhull) [inline src/quo2/foundations/typography.cljs:23]: I am not sure if we need label to be uppercase by default? ([figma](https://www.figma.com/file/v98g9ZiaSHYUdKWrbFg9eM/Mobile-Foundations?node-id=625%3A38))

File: translations/en.json
Comment (flexsurfer) [inline translations/en.json:989]: could you elaborate on why we need dapp2 ?

File: src/quo2/foundations/typography.cljs
Comment (flexsurfer) [reply src/quo2/foundations/typography.cljs:23]: at least i was when i implementing it

**Comment by Parveshdhull** [comment]
Looks like component is not perfectly aligning with design, due to different letter spacing

shot1
shot3

File: src/quo2/components/information_box.cljs
Comment (flexsurfer) [inline src/quo2/components/information_box.cljs]: im not sure async-storage should be used  inside component

**Review by flexsurfer — APPROVED** [review]
nice! great job!

File: translations/en.json
Comment (Parveshdhull) [reply translations/en.json:989]: Hi, Thanks for the PR review.
According to the design, the action button should have an `Open dApp` label. But currently, only`Open ÐApp` label is available.
image

File: src/quo2/components/information_box.cljs
Comment (Parveshdhull) [reply src/quo2/components/information_box.cljs]: any suggestions?
I guess we can pass this call as `on-close` while creating the component, but this will be the same for all closable information boxes and having this call inside the component encapsulates its functionality.

Comment (Parveshdhull) [reply src/quo2/components/information_box.cljs]: yup, I also don't like having async-storage in component. maybe we can create a file `quo2.core` or `quo2.components.core` where we load states of all information box components on start and dispatch event for storing on on-close. wdyt?

Comment (flexsurfer) [reply src/quo2/components/information_box.cljs]: i don't think we'll have lots of states, so we could just manage state locally in the component parent view

File: src/quo2/foundations/typography.cljs
Comment (Parveshdhull) [inline src/quo2/foundations/typography.cljs:3]: this can't be the formula for letter-spacing, as it is inversely proportional,
when there should be more spacing it will give less, when there should be less like 0, it will give infinite

File: src/quo2/components/information_box.cljs
Comment (flexsurfer) [inline src/quo2/components/information_box.cljs]: it would be better to abstract the component completely, so we need to pass closed as a parameter from outside the component

File: src/quo2/foundations/typography.cljs
Comment (hesterbruikman) [reply src/quo2/foundations/typography.cljs:3]: [UPDATED] ~~@Parveshdhull [Figma](https://forum.figma.com/t/letter-spacing-should-not-be-percentage-based/3062) can also express the letter spacing in `em` rather than percentages. The body text having a spacing of `-0.003em` cc @pedro-et~~

There's a formula provided for the Inter font here: https://rsms.me/inter/dynmetrics/ (formula for letter-spacing which is 'tracking'). As far as I can tell we haven't deviated from this formula and you should be able to use it to set the letter-spacing. I'll leave it to @pedro-et to confirm this though

Comment (Parveshdhull) [reply src/quo2/foundations/typography.cljs:3]: hi @hesterbruikman, Thank you very much for the link, I will explore the program/formula and see how it can be used.

**Review by flexsurfer — APPROVED** [review]

**Comment by qoqobolo** [comment]
@Parveshdhull thanks for your great work!
Couldn't find any issues. 
Checked:
- upgrade from 1.19 to this PR (Android)
- fresh installation (without upgrade)
- new/recovered multi-account
- multi-account with/without ENS
- regular/keycard multi-account
- several multi-accounts on one device
- dark/light mode
- iOS
- with chats on the home screen
- search field on the home screen
- with communities enabled

One request: can you add an accessibility-id for this button, please? We'll need it for e2e fixes. 
Screenshot 2022-08-11 at 17 04 12

After this, PR can be merged.
Thanks!

**Comment by Parveshdhull** [comment]
> One request: can you add an accessibility-id for this button, please? We'll need it for e2e fixes.

sure will do, thank you

**Pull request merged**
